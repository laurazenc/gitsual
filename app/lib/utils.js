// @flow

export const pick = (obj: T, paths: K[]): Pick<T, K> => {
  return {
    ...paths.reduce((mem, key) => ({ ...mem, [key]: obj[key] }), {})
  };
};
