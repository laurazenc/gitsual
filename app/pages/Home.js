import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { findProject, listProject } from '../store/actions/project';

const Home = ({ projects, actions }) => {
  console.log(actions);
  useEffect(() => {
    const getProjects = async () => {
      await actions.listProject();
    };
    getProjects();
  }, []);

  const retrieveProject = async () => {
    await actions.findProject();
  };

  return (
    <div>
      Home
      <button onClick={() => retrieveProject()}>Find project</button>
      <div>Current repos:</div>
      {projects.map(project => {
        return (
          <Link key={project.name} to={`/repo/${project.name}`}>
            {project.name}
          </Link>
        );
      })}
    </div>
  );
};

Home.propTypes = {
  projects: PropTypes.array,
  actions: PropTypes.object
};

const mapStateToProps = ({ project, repo }) => ({
  projects: project.list,
  currentRepo: project.repo,
  repo
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ findProject, listProject }, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
