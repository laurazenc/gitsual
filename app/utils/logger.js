// @gflow
import winston, { createLogger, format, transports } from 'winston';
import config from '../config';

const { combine, timestamp, label, prettyPrint } = format;

const logger = createLogger({
  level: config.logs.level,
  levels: winston.config.npm.levels,
  format: combine(label(), timestamp(), prettyPrint()),
  transports: [new transports.Console()]
});

export default logger;
