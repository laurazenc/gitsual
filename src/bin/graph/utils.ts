export const pick = (obj: T, paths: K[]): Pick<T, K> => {
    return {
        ...paths.reduce((mem, key) => ({ ...mem, [key]: obj[key] }), {}),
    }
}

export const describeArc = (
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    radius: number,
    startAngle: number,
    endAngle: number,
) => {
    const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1'

    const d = [
        'M',
        x1,
        y1 - radius,
        'L',
        x1,
        y2 + radius,
        'A',
        radius,
        radius,
        0,
        largeArcFlag,
        0,
        x1 - radius,
        y2,
        'L',
        x2 + radius,
        y2,
    ].join(' ')

    return d
}

export const describeInverseArc = (
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    radius: number,
    startAngle: number,
    endAngle: number,
) => {
    const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1'

    const d = [
        'M',
        x2 + radius,
        y2,
        'L',
        x1 - radius,
        y2,
        'A',
        radius,
        radius,
        0,
        largeArcFlag,
        0,
        x1,
        y2 - radius,
        'L',
        x1,
        y1 + radius,
    ].join(' ')

    return d
}

export const describeInverseRightArc = (
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    radius: number,
    startAngle: number,
    endAngle: number,
) => {
    const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1'

    const d = [
        'M',
        x1 - radius,
        y1,
        'L',
        x2 + radius,
        y1,
        'A',
        radius,
        radius,
        0,
        largeArcFlag,
        0,
        x2,
        y1 + radius,
        'L',
        x2,
        y2 + radius,
    ].join(' ')

    return d
}
export const describeInverseLeftArc = (
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    radius: number,
    startAngle: number,
    endAngle: number,
) => {
    const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1'
    const d = [
        'M',
        x1,
        y1,
        'L',
        x1,
        y2 - radius,
        'A',
        radius,
        radius,
        0,
        largeArcFlag,
        0,
        x1 + radius,
        y2,
        'L',
        x2 - radius,
        y2,
    ].join(' ')

    return d
}

export const drawLine = (x1: number, y1: number, x2: number, y2: number) => {
    const d = ['M', x1, y1, 'L', x2, y2].join(' ')

    return d
}
