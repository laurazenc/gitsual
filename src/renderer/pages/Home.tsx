import * as React from 'react'
import { connect, ConnectedProps } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import styled from 'styled-components'
import { AnyAction } from 'redux'
import { Link } from 'react-router-dom'

import { Button, Flex, Icon, Text } from '../components'
import { AppState } from '../store/reducers'
import { Project } from '../store/reducers/projects'
import { openProject, loadProjects, ProjectActions } from '../store/actions/projects'

const FlexContainer = styled(Flex)`
    height: 100%;
`

const Path = styled(Text)`
    text-overflow: ellipsis;
    overflow: hidden;
`

const Panel = styled.div``

interface HomeProps {
    openProject: () => void
    loadProjects: () => void
    recentProjects: Project[] | any
}

const Home: React.FC<HomeProps> = ({ recentProjects, loadProjects, openProject }) => {
    const open = async () => {
        await openProject()
        await loadProjects()
    }

    return (
        <FlexContainer justify="space-between" align="center" direction="column" padding="16px">
            <Flex direction="column" align="center" justify="center" maxWidth="700px" style={{ height: '100%' }}>
                <Icon name="logo" />
                <Text size="21px" height="30px" margin="30px 0">
                    Welcome to gitsual
                </Text>
                <Flex justify="space-between" margin="100px 0 0">
                    <Panel>
                        <Button
                            border={true}
                            radius="3px"
                            justify="flex-start"
                            margin="0 0 10px 0"
                            padding="6px 12px"
                            onClick={() => open()}
                        >
                            <Icon name="folder" color="accent" />
                            <Text size="16px" height="24px" margin="0 0 0 10px">
                                Open existing repo
                            </Text>
                        </Button>
                        <Button border={true} radius="3px" justify="flex-start" margin="0 0 10px 0" padding="6px 12px">
                            <Icon name="laptop" color="accent" />
                            <Text size="16px" height="24px" margin="0 0 0 10px">
                                Clone repository
                            </Text>
                        </Button>
                        <Button border={true} radius="3px" justify="flex-start" padding="6px 12px">
                            <Icon name="file" color="accent" />
                            <Text size="16px" height="24px" margin="0 0 0 10px">
                                Start new repo
                            </Text>
                        </Button>
                    </Panel>
                    {recentProjects.length > 0 && (
                        <Panel>
                            <Text size="20px" height="30px" margin="0 0 16px 0">
                                Recent
                            </Text>
                            {recentProjects.map((project: Project, i: number) => {
                                return (
                                    <Flex key={i} maxWidth="250px">
                                        <Link to={`/repo/${project.name}`}>
                                            <Text size="16px" height="24px" margin="0 12px 0 0" color="accent">
                                                {project.name}
                                            </Text>
                                        </Link>
                                        <Path size="16px" height="24px" color="light">
                                            {project.path}
                                        </Path>
                                    </Flex>
                                )
                            })}
                        </Panel>
                    )}
                    <Panel>
                        <Text size="20px" height="30px" margin="0 0 16px 0">
                            Customize
                        </Text>
                        <Flex align="center">
                            <Icon name="cog" />
                            <Text size="16px" height="24px" color="accent" margin="0 0 0 10px">
                                Preferences
                            </Text>
                        </Flex>
                    </Panel>
                </Flex>
            </Flex>
            <Flex justify="center">
                <Text size="12px" height="16px">
                    Made with{' '}
                </Text>
                <Icon name="heart" color="red" />
                <Text size="12px" height="16px">
                    {' '}
                    by{' '}
                </Text>
                <Text size="12px" height="16px" color="accent">
                    @laurazenc
                </Text>
            </Flex>
        </FlexContainer>
    )
}

const mapState = (state: AppState) => ({
    recentProjects: state.projects.projects,
})

const mapDispatch = (dispatch: ThunkDispatch<any, any, AnyAction>) => ({
    openProject: () => dispatch(openProject()),
    loadProjects: () => dispatch(loadProjects()),
})

export default connect(mapState, mapDispatch)(Home)
