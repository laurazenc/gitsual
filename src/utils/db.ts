import fs from 'fs'
import path from 'path'

try {
    if (!fs.existsSync(path.resolve(`./.gitsual`))) fs.mkdirSync(path.resolve(`./.gitsual`))
} catch (ignore) { } // eslint-disable-line

export const DB_PATH = path.resolve(`./.gitsual/db.json`) // eslint-disable-line