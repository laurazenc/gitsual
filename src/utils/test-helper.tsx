import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { createStore, AnyAction, Middleware } from 'redux'
import { BrowserRouter } from 'react-router-dom'

import configureStore, { MockStoreEnhanced, MockStoreCreator } from 'redux-mock-store'
import thunk, { ThunkDispatch } from 'redux-thunk'

import { defaultTheme } from '../shared/theme'
import { createHashHistory } from 'history'

import { initialState as projectsInitState } from '../renderer/store/reducers/projects'
import { initialState as themeInitState } from '../renderer/store/reducers/theme'

const initialState = {
    projects: projectsInitState,
    theme: themeInitState,
}
const middlewares: Array<Middleware> = [thunk]
type DispatchExts = ThunkDispatch<any, undefined, AnyAction>
const mockStoreCreator: MockStoreCreator<any, DispatchExts> = configureStore<any, DispatchExts>(middlewares)

interface Options {
    mocked?: boolean
    customState?: any
}

const customRender = (ui: React.ReactElement<any>, options: Options = {}) => {
    const { mocked = false, customState = null } = options

    const mockStore: MockStoreEnhanced<any, DispatchExts> = mockStoreCreator(
        mocked ? Object.assign(initialState, customState) : initialState,
    )

    return {
        store: mockStore,
        ...render(
            <Provider store={mockStore}>
                <ThemeProvider theme={defaultTheme}>
                    <BrowserRouter>{ui}</BrowserRouter>
                </ThemeProvider>
            </Provider>,
        ),
    }
}

export * from '@testing-library/react'

export { customRender as render }
