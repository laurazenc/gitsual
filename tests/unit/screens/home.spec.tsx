import React from 'react'
import { render, wait } from '../../../src/utils/test-helper'

import Home from '../../../src/renderer/pages/Home'

describe('Home screen', () => {
    test('should render', () => {
        const { container } = render(<Home />)
        expect(container).toMatchSnapshot()
    })

    test('should not render the recent projects columns if there is no loaded project ', async () => {
        const { queryAllByText } = render(<Home />)
        const column = await wait(() => queryAllByText('Recent'))
        expect(column).toBeUndefined()
    })

    test('should render the recent projects columns', async () => {
        const { queryAllByText } = render(<Home />, {
            mocked: true,
            customState: { projects: { projects: [{ name: 'wadus', path: 'path' }] } },
        })

        const [projectName] = queryAllByText('wadus')
        expect(projectName).toBeDefined()
    })
})
