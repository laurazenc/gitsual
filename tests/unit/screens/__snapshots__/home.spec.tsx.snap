// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should render 1`] = `
<div>
  <div
    class="sc-bxivhb sc-ifAKCX hVKjLB"
    direction="column"
  >
    <div
      class="sc-bxivhb jFLSFA"
      direction="column"
      style="height: 100%;"
    >
      <svg
        fill="none"
        height="72"
        viewBox="0 0 148 72"
        width="148"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M73.898 0.0006464C65.8076 0.0519516 58.441 3.15429 52.8911 8.21493L52.9409 8.26478C52.3954 8.74227 50.5861 10.1858 47.8575 10.9586C46.2073 11.4259 44.3694 11.3108 42.9439 11.0788C31.7621 6.7212 18.5755 8.9848 9.44809 17.9305C-2.97242 30.1038 -3.17283 50.0411 9.00047 62.4616C21.1738 74.8821 41.111 75.0825 53.5315 62.9092C61.8996 54.7077 64.7208 42.9821 61.9482 32.4054L61.9708 32.4C61.9708 32.4 60.6564 28.3205 61.6575 25.0388C62.6473 21.7943 64.5701 19.8841 64.6135 19.8412L64.6143 19.8405L64.5885 19.8141C67.1575 17.7183 70.4313 16.4535 73.9997 16.4298C77.5684 16.4534 80.8424 17.7182 83.4116 19.8141L83.3857 19.8405C83.3858 19.8405 85.3414 21.7572 86.3425 25.0389C87.3437 28.3206 86.0292 32.4001 86.0292 32.4001L86.0518 32.4054C83.2792 42.9821 86.1004 54.7077 94.4685 62.9092C106.889 75.0825 126.826 74.8821 139 62.4616C151.173 50.0411 150.972 30.1039 138.552 17.9305C129.424 8.98482 116.238 6.72122 105.056 11.0788C103.631 11.3108 101.793 11.426 100.143 10.9586C97.4139 10.1858 95.6046 8.74225 95.0591 8.26477L95.1089 8.21493C89.559 3.15429 82.1925 0.0519516 74.102 0.000646401V0L74 0.00016156L73.898 0V0.0006464ZM105.754 29.8782C111.576 23.9379 121.112 23.8421 127.052 29.6641C132.992 35.4861 133.088 45.0213 127.266 50.9616C121.444 56.9018 111.909 56.9977 105.968 51.1756C100.028 45.3536 99.9324 35.8184 105.754 29.8782ZM42.2456 29.8782C36.4236 23.9379 26.8884 23.8421 20.9481 29.6641C15.0079 35.4861 14.912 45.0213 20.734 50.9615C26.5561 56.9018 36.0912 56.9976 42.0315 51.1756C47.9717 45.3536 48.0676 35.8184 42.2456 29.8782Z"
          fill="url(#paint0_linear)"
          fill-rule="evenodd"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="paint0_linear"
            x1="143.369"
            x2="131.263"
            y1="1.00631"
            y2="84.7461"
          >
            <stop
              stop-color="#F6D2B0"
            />
            <stop
              offset="1"
              stop-color="#FF406F"
            />
          </lineargradient>
        </defs>
      </svg>
      <div
        class="sc-bwzfXH crCCPa"
        height="30px"
      >
        Welcome to gitsual
      </div>
      <div
        class="sc-bxivhb eYnCW"
      >
        <div
          class="sc-bZQynM bOrPFB"
        >
          <div
            class="sc-htpNat dhLikq"
            radius="3px"
          >
            <svg
              class="sc-bdVaJa huzVGd"
              color="accent"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/Svg"
            >
              <g
                id="folder-opened"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.6666 3.33333C13.403 3.33333 14 3.93029 14 4.66667V5.33333C14.7363 5.33333 15.3333 5.93029 15.3333 6.66667L15.3174 6.81129L13.9973 12.7519C13.9533 13.4486 13.3744 14 12.6666 14H1.99996C1.26358 14 0.666626 13.403 0.666626 12.6667V3.33333C0.666626 2.59695 1.26358 2 1.99996 2H5.99996C6.74663 2 7.22192 2.31697 7.69928 2.91924C7.71247 2.93589 7.74462 2.97731 7.78022 3.02319L7.78034 3.02335L7.78042 3.02345C7.83071 3.08826 7.88785 3.16189 7.90828 3.18721C7.92894 3.21283 7.94637 3.23402 7.96076 3.25153L7.96077 3.25154C8.01941 3.32285 8.02791 3.33319 8.00115 3.33333H12.6666ZM12.6666 4.66667V5.33333H3.33329C2.55556 5.33333 2.20676 5.74764 2.01938 6.50694L1.99996 6.59431V3.33333H5.99996C6.25612 3.33333 6.40464 3.43238 6.65436 3.74744C6.66221 3.75734 6.68471 3.78634 6.71274 3.82248C6.76633 3.89156 6.84014 3.9867 6.87051 4.02436C7.20575 4.43994 7.51762 4.66409 7.99639 4.66666L12.6666 4.66667ZM12.6666 12.6667H2.01622L3.31741 6.81129C3.33323 6.74753 3.34707 6.6997 3.35749 6.66667H13.9837L12.6825 12.522L12.6666 12.6667Z"
                  fill="accent"
                  fill-rule="evenodd"
                  id="icon"
                />
              </g>
            </svg>
            <div
              class="sc-bwzfXH fnaits"
              height="24px"
            >
              Open existing repo
            </div>
          </div>
          <div
            class="sc-htpNat dhLikq"
            radius="3px"
          >
            <svg
              class="sc-bdVaJa huzVGd"
              color="accent"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="laptop"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.6666 2.66667H3.33329C2.59691 2.66667 1.99996 3.26363 1.99996 4.00001V10H0.666626V12C0.666626 12.7364 1.26358 13.3333 1.99996 13.3333H14C14.7363 13.3333 15.3333 12.7364 15.3333 12V10H14V4.00001C14 3.26363 13.403 2.66667 12.6666 2.66667ZM12.6666 10H3.33329V4.00001H12.6666V10ZM1.99996 12V11.3333H14V12H1.99996Z"
                  fill="accent"
                  fill-rule="evenodd"
                  id="icon"
                />
              </g>
            </svg>
            <div
              class="sc-bwzfXH fnaits"
              height="24px"
            >
              Clone repository
            </div>
          </div>
          <div
            class="sc-htpNat kUOXSM"
            radius="3px"
          >
            <svg
              class="sc-bdVaJa huzVGd"
              color="accent"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="file"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.66667 1.99999H3.33333V14H10V15.3333H3.33333C2.59695 15.3333 2 14.7364 2 14V1.99999C2 1.26361 2.59695 0.666656 3.33333 0.666656H10.2761L14 4.39051V9.33332H12.6667V5.99999H10C9.26362 5.99999 8.66667 5.40304 8.66667 4.66666V1.99999ZM12.6667 12.6667V11.3333H14V12.6667H15.3333V14H14V15.3333H12.6667V14H11.3333V12.6667H12.6667ZM10 2.27613L12.3905 4.66666H10V2.27613Z"
                  fill="accent"
                  fill-rule="evenodd"
                  id="icon"
                />
              </g>
            </svg>
            <div
              class="sc-bwzfXH fnaits"
              height="24px"
            >
              Start new repo
            </div>
          </div>
        </div>
        <div
          class="sc-bZQynM bOrPFB"
        >
          <div
            class="sc-bwzfXH SBoMB"
            height="30px"
          >
            Customize
          </div>
          <div
            class="sc-bxivhb bzyTmT"
          >
            <svg
              class="sc-bdVaJa jWsai"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="cog"
              >
                <path
                  clip-rule="evenodd"
                  d="M3.95125 12.0488L5.07422 15.0635L8.00001 13.7258L10.9258 15.0635L12.0488 12.0488L15.0635 10.9258L13.7258 8L15.0635 5.07421L12.0488 3.95124L10.9258 0.936508L8.00001 2.27418L5.07422 0.936508L3.95125 3.95124L0.936523 5.07421L2.2742 8L0.936523 10.9258L3.95125 12.0488ZM10.1767 13.2549L8.00001 12.2597L5.82336 13.2549L4.98792 11.0121L2.74511 10.1767L3.74027 8L2.74511 5.82335L4.98792 4.98791L5.82336 2.7451L8.00001 3.74026L10.1767 2.7451L11.0121 4.98791L13.2549 5.82335L12.2597 8L13.2549 10.1767L11.0121 11.0121L10.1767 13.2549ZM8.00002 11.3333C6.15907 11.3333 4.66669 9.84095 4.66669 8C4.66669 6.15905 6.15907 4.66667 8.00002 4.66667C9.84097 4.66667 11.3334 6.15905 11.3334 8C11.3334 9.84095 9.84097 11.3333 8.00002 11.3333ZM10 8C10 9.10457 9.10459 10 8.00002 10C6.89545 10 6.00002 9.10457 6.00002 8C6.00002 6.89543 6.89545 6 8.00002 6C9.10459 6 10 6.89543 10 8Z"
                  fill-rule="evenodd"
                  id="icon"
                />
              </g>
            </svg>
            <div
              class="sc-bwzfXH fmocfR"
              color="accent"
              height="24px"
            >
              Preferences
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bxivhb dXvdhZ"
    >
      <div
        class="sc-bwzfXH jmlWCa"
        height="16px"
      >
        Made with
         
      </div>
      <svg
        class="sc-bdVaJa MfkVK"
        color="red"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="heart"
        >
          <path
            clip-rule="evenodd"
            d="M11.0354 1.33332C10.0703 1.33331 9.44822 1.44829 8.73506 1.79736C8.47103 1.9266 8.22332 2.08224 7.99327 2.26382C7.77209 2.09242 7.53428 1.94393 7.28117 1.81874C6.55417 1.45916 5.90015 1.33331 4.96966 1.33331C2.47758 1.33331 0.666626 3.39816 0.666626 6.0799C0.666626 8.10515 1.79585 10.0615 3.89919 11.9532C5.00322 12.9462 6.41284 13.9289 7.42236 14.4516L7.99996 14.7507L8.57756 14.4516C9.58708 13.9289 10.9967 12.9462 12.1007 11.9532C14.2041 10.0615 15.3333 8.10515 15.3333 6.0799C15.3333 3.42654 13.5053 1.34355 11.0354 1.33332ZM14 6.0799C14 7.66658 13.0574 9.29952 11.2091 10.9619C10.2014 11.8682 8.90593 12.7739 7.99996 13.2491C7.09399 12.7739 5.79851 11.8682 4.79081 10.9619C2.94251 9.29952 1.99996 7.66658 1.99996 6.0799C1.99996 4.09846 3.25572 2.66665 4.96966 2.66665C5.70948 2.66665 6.16538 2.75437 6.69005 3.01388C6.99996 3.16716 7.27388 3.36956 7.51098 3.62249L8.00152 4.14578L8.48759 3.61835C8.73003 3.35528 9.00797 3.14827 9.32124 2.99493C9.83054 2.74564 10.258 2.66665 11.0327 2.66665C12.7254 2.67367 14 4.12596 14 6.0799Z"
            fill="#FF406F"
            fill-rule="evenodd"
            id="icon"
          />
        </g>
      </svg>
      <div
        class="sc-bwzfXH jmlWCa"
        height="16px"
      >
         
        by
         
      </div>
      <div
        class="sc-bwzfXH cprfsj"
        color="accent"
        height="16px"
      >
        @laurazenc
      </div>
    </div>
  </div>
</div>
`;
